metalearners/metalearner_grid_search_cv.py

```diff
@@ -0,0 +1,296 @@
+# Copyright (c) QuantCo 2024-2024
+# SPDX-License-Identifier: BSD-3-Clause
+
+import time
+from collections.abc import Callable, Mapping, Sequence
+from dataclasses import dataclass
+from functools import reduce
+from operator import add
+
+import pandas as pd
+from joblib import Parallel, delayed
+from sklearn.model_selection import KFold, ParameterGrid
+
+from metalearners._typing import Matrix, OosMethod, Vector, _ScikitModel
+from metalearners._utils import index_matrix, index_vector
+from metalearners.cross_fit_estimator import OVERALL
+from metalearners.metalearner import PROPENSITY_MODEL, MetaLearner
+
+
+@dataclass(frozen=True)
+class _FitAndScoreJob:
+    metalearner: MetaLearner
+    X_train: Matrix
+    y_train: Vector
+    w_train: Vector
+    X_test: Matrix
+    y_test: Vector
+    w_test: Vector
+    oos_method: OosMethod
+    scoring: Mapping[str, list[str | Callable]] | None
```

Comment:
We have type hint variables for this in `_typing`. :)

Reply:
[c08dd6a](https://github.com/Quantco/metalearners/pull/9/commits/c08dd6ae69dd38cdfb96ec2eea78bc40dad631e9)